google/cloud/bigquery/job/base.py

```diff
@@ -659,6 +660,12 @@ def __init__(self, job_type, **kwargs):
         for prop, val in kwargs.items():
             setattr(self, prop, val)
 
+    def __setattr__(self, name, value):
+        """Override to be able to warn if an uknown property is being set"""
+        if not name.startswith("_") and name not in type(self).__dict__:
+            warnings.warn("Property {} is unknown for {}.".format(name, type(self)))
```

Comment:
On second thought, I think we can go a step further and throw `AttributeError`, just like `object()` instances do.

Reply:
OK. Done.